cs/CMakeLists.txt

```diff
@@ -118,6 +118,7 @@ set(BASE_SOURCES
   ${BASE_DIR}/printer.cpp
   ${BASE_DIR}/release_string.c
   ${BASE_DIR}/resource_tracker.cpp
+  ${BASE_DIR}/server_type.cpp
```

Comment:
`cs` is the client library. server_type and connection_server_rules belong to server module only. Adding the files here is not necessary, please remove them.